heaps/min_heap.py

```diff
@@ -17,62 +17,97 @@ class MinHeap:
     def __init__(self):
         self.store = []
 
-
-    def add(self, key, value = None):
+    def add(self, key, value=None):
         """ This method adds a HeapNode instance to the heap
             If value == None the new node's value should be set to key
-            Time Complexity: ?
-            Space Complexity: ?
+            Time Complexity: O(logn)
+            Space Complexity: O(1)
         """
-        pass
+        if value is None:
+            value = key
+        # creating a heap node instance and appending to store
+        self.store.append(HeapNode(key, value))
+        # moving pointer
+        self.heap_up(len(self.store) - 1)
 
     def remove(self):
         """ This method removes and returns an element from the heap
             maintaining the heap structure
-            Time Complexity: ?
-            Space Complexity: ?
+            Time Complexity: O(logn)
+            Space Complexity: O(1)
```

Comment:
👍 , however the space complexity is O(log n) because of the recursive call stack.